terratest/test/ohmyglb_basic_split_failover_test.go

```diff
@@ -0,0 +1,115 @@
+	require.NoError(t, err)
+
+	// To ensure we can reuse the resource config on the same cluster to test different scenarios, we setup a unique
+	// namespace for the resources for this test.
+	// Note that namespaces must be lowercase.
+	namespaceName := fmt.Sprintf("ohmyglb-test-%s", strings.ToLower(random.UniqueId()))
+
+	// Here we choose to use the defaults, which is:
+	// - HOME/.kube/config for the kubectl config file
+	// - Current context of the kubectl config file
+	// - Random namespace
+	optionsContext1 := k8s.NewKubectlOptions("kind-test-gslb1", "", namespaceName)
+	optionsContext2 := k8s.NewKubectlOptions("kind-test-gslb2", "", namespaceName)
+
+	k8s.CreateNamespace(t, optionsContext1, namespaceName)
+	k8s.CreateNamespace(t, optionsContext2, namespaceName)
+	defer k8s.DeleteNamespace(t, optionsContext1, namespaceName)
+	defer k8s.DeleteNamespace(t, optionsContext2, namespaceName)
+
+	gslbName := "test-gslb"
+
+	createGslbWithHealthyApp(t, optionsContext1, kubeResourcePath1, gslbName)
+
+	createGslbWithHealthyApp(t, optionsContext2, kubeResourcePath2, gslbName)
+
+	expectedIPsCluster1 := GetIngressIPs(t, optionsContext1, gslbName)
+	expectedIPsCluster2 := GetIngressIPs(t, optionsContext2, gslbName)
+
+	t.Run("Each cluster resolves its own set of IP addresses", func(t *testing.T) {
+		beforeFailoverResponseCluster1, err := DoWithRetryWaitingForValueE(
+			t,
+			"Wait 1st cluster coredns to pickup dns values...",
+			300,
+			1*time.Second,
+			func() ([]string, error) { return Dig(t, "localhost", 53, "terratest-failover.cloud.example.com") },
+			expectedIPsCluster1)
+		require.NoError(t, err)
+
+		assert.Equal(t, beforeFailoverResponseCluster1, expectedIPsCluster1)
+
+		beforeFailoverResponseCluster2, err := DoWithRetryWaitingForValueE(
+			t,
+			"Wait 2nd cluster coredns to pickup dns values...",
+			300,
+			1*time.Second,
+			func() ([]string, error) { return Dig(t, "localhost", 54, "terratest-failover.cloud.example.com") },
+			expectedIPsCluster2)
+		require.NoError(t, err)
+
+		assert.Equal(t, beforeFailoverResponseCluster2, expectedIPsCluster2)
+	})
+
+	t.Run("serviceHealth becomes Unhealthy after scaling down to 0", func(t *testing.T) {
+
+		k8s.RunKubectl(t, optionsContext1, "scale", "deploy", "frontend-podinfo", "--replicas=0")
+
+		assertGslbStatus(t, optionsContext1, gslbName, "terratest-failover.cloud.example.com:Unhealthy")
+	})
+
+	t.Run("Cluster 1 failovers to Cluster 2", func(t *testing.T) {
+
+		afterFailoverResponse, err := DoWithRetryWaitingForValueE(
+			t,
+			"Wait for failover to happen and coredns to pickup new values(cluster1)...",
+			300,
+			1*time.Second,
+			func() ([]string, error) { return Dig(t, "localhost", 53, "terratest-failover.cloud.example.com") },
+			expectedIPsCluster2)
+		require.NoError(t, err)
+
+		assert.Equal(t, afterFailoverResponse, expectedIPsCluster2)
+	})
+
+	t.Run("Cluster 2 still returns own entries", func(t *testing.T) {
```

Comment:
@ytsarev would it be worth adding an additional step. Test if the service from Cluster 1 becomes healthy once more, that  Cluster 1 should then continue to resolve its own IP addresses? I.e. it fails back again.

So:

1. Cluster 1 resolves it's own addresses
2. Service becomes unhealthy on Cluster 1
3. Cluster 2 addresses are now resolved only
4. Service becomes healthy again on Cluster 1
5. Cluster 1 once again resolves it's own addresses

Reply:
totally, will add this step, thanks!

Reply:
@donovanmuller implemented, please review